protostar/commands/cairo1_commands/cairo1_build_command.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from pathlib import Path
+from typing import List, Optional, Any
+
+from protostar.cli import ProtostarArgument, ProtostarCommand
+from protostar.cli.common_arguments import COMPILED_CONTRACTS_DIR_ARG
+from protostar.compiler import (
+    ProjectCompiler,
+)
+import protostar.cairo.cairo_bindings as cairo1
+
+
+class Cairo1BuildCommand(ProtostarCommand):
+    def __init__(
+        self,
+        project_compiler: ProjectCompiler,
+    ):
+        super().__init__()
+        self._project_compiler = project_compiler
+
+    @property
+    def example(self) -> Optional[str]:
+        return "$ protostar build-cairo1"
+
+    @property
+    def name(self) -> str:
+        return "build-cairo1"
+
+    @property
+    def description(self) -> str:
+        return "Compile cairo1 contracts."
+
+    @property
+    def arguments(self):
+        return [
+            ProtostarArgument(
+                name="cairo-path",
+                description="Additional directories to look for sources.",
+                type="path",
+                value_parser="list",
+            ),
+            COMPILED_CONTRACTS_DIR_ARG,
+        ]
+
+    async def run(self, args: Any):
+        logging.info("Building cairo1 contracts")
+        try:
+            await self.build(
+                output_dir=args.compiled_contracts_dir,
+                relative_cairo_path=args.cairo_path,
+            )
+        except BaseException as ex:
+            logging.error("Build failed")
+            raise ex
+
+        logging.info("Contracts built successfully")
+
+    async def build(
+        self,
+        output_dir: Path,
+        relative_cairo_path: Optional[List[Path]] = None,
+    ) -> None:
+        configuration_file = self._project_compiler.configuration_file
+        cairo_path = relative_cairo_path or []
+        for contract_name in configuration_file.get_contract_names():
```

Comment:
This will compile all contracts as cairo1 contracts, which will be wrong for most of the projects

Reply:
My understanding was that if someone calls `build-cairo1`, they want to build cairo1 contracts. So I assume that now projects may contain mixed contracts - some of them will be written in cairo 0 and some of them in cairo 1, right?

Could we do this as a separate task, especially since I think some discussions regarding this are in progress?

Reply:
taking into account everything we discussed, i think we should leave behavior as-is, if users want to build cairo0 contacts they will use the legacy command